content/blog/model-calibration/index.Rmd

```diff
@@ -0,0 +1,409 @@
+
+```{r, fig.alt = "Plot illustrating the horizontal location of each step and the size of the window"}
+#| echo: false
+#| fig.width: 6
+#| fig.height: 4
+#| out.width: 70%
+windows_size <- 0.10
+step_size    <- 0.05
+
+steps <- seq(0, 1, by = step_size)
+lower_cut <- steps - (windows_size / 2)
+lower_cut[lower_cut < 0] <- 0
+upper_cut <- steps + (windows_size / 2)
+upper_cut[upper_cut > 1] <- 1
+
+tibble(
+  step = steps,
+  lower = lower_cut,
+  upper = upper_cut,
+  group = seq_along(steps)
+) %>% 
+  ggplot(aes(x = step, y = group)) + 
+  geom_errorbar(aes(xmin = lower, xmax = upper)) +
+  labs(y = NULL, x = "predicted probability") +
+  theme_minimal()
+```
+
+In probably, the `cal_plot_windowed()` function provides this functionality. 
+The default step size is 0.05, and can be changed via the `step_size` argument.
+The default window size is 0.1, and can be changed via the `window_size` argument.
+
+```{r, fig.alt = "Calibration plot with 21 windows, created with the cal_plot_windowed() function"}
+segment_logistic %>% 
+  cal_plot_windowed(Class, .pred_good)
+```
+
+Here is an example of reducing the `step_size` from 0.05, to 0.02. There are
+more than double the windows:
+
+```{r, fig.alt = "Calibration plot with more steps than the default, created with the cal_plot_windowed() function"}
+segment_logistic %>% 
+  cal_plot_windowed(Class, .pred_good, step_size = 0.02)
+```
+
+### Model-Based
+
+Another way to visualize the performance is to fit a classification model of the
+events against the estimated probabilities. This is helpful because it avoids the use of
+pre-determined groupings. Another difference, is that we are not plotting 
+midpoints of actual results, but rather predictions based on those results. 
+
+The `cal_plot_logistic()` provides this functionality. By default, it uses a 
+logistic regression. There are two possible methods for fitting: 
+
+spline model, provided by the `mgcv` package. The idea is to visualize
+a smooth line based on the predictions based on the `smooth` argument:
+
+* `smooth = TRUE` (the default) fits a generalized additive model using splines. This allows for more flexible model fits. 
+* `smooth = FALSE` uses an ordinary logistic regression model with linear terms for the predictor.
+
+As an example: 
+
+```{r, fig.alt = "Logistic Spline calibration plot, created with the cal_plot_logistic() function"}
+segment_logistic %>% 
+  cal_plot_logistic(Class, .pred_good)
+```
+
+The cooresponding `glm()` model produces:
+
+```{r, fig.alt = "Ordinary logistic calibration plot, created with the cal_plot_logistic() function"}
+segment_logistic %>% 
+  cal_plot_logistic(Class, .pred_good, smooth = FALSE)
+```
+
+### Additional options and features
+
+#### Intervals
```

Comment:
In the deploy preview, it's hard to tell the difference between the L3 header above and this L4 one. Maybe we can delete the L4 headers and inline bold the first mentions of **interval**s and **rugs**?